### PR TITLE
[TR] [SLA] PIM-5257: Increase precision of data and baseData in Metric entity

### PR DESCRIPTION
**Pull request by willy-ahva**
| Q                 | A
| ----------------- | ---
| Specs             | ok
| Behats            | ok
| Blue CI           | running
| Changelog updated | yes
| Review and 2 GTM  |

Add precision to the Metric entity to be able to save data and baseData for number under 1E-4.
We need this precision, for example, when we convert data in millimeter cube to meter cube (1E-8).
For data which are already saved, as many 0 as needed will be added at the end of the number.

Need a GTM from @BitOne

### CODE REVIEW

**Comment by jjanvier** [comment]
no behat, not gtm
impossible to add a real use case?

File: upgrades/schema/Version20151202101336.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Pim\Upgrade\Schema;
+
+use Doctrine\DBAL\Migrations\AbstractMigration;
+use Doctrine\DBAL\Schema\Schema;
+
+/**
+ * Add precision to the Metric entity to be able to save data and baseData for number under 1E-4.
```
Comment (rybus) [inline upgrades/schema/Version20151202101336.php]: Adds [...] for numbers

**Comment by ClementGautier** [comment]
behats have been added

File: upgrades/schema/Version20151202101336.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+class Version20151202101336 extends AbstractMigration
+{
+    /**
+     * @param Schema $schema
+     */
+    public function up(Schema $schema)
+    {
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() != 'mysql', 'Migration can only be executed safely on \'mysql\'.');
+
+        $this->addSql('ALTER TABLE pim_catalog_metric CHANGE data data NUMERIC(14, 10) DEFAULT NULL, CHANGE base_data base_data NUMERIC(14, 10) DEFAULT NULL');
```
Comment (skeleton) [inline upgrades/schema/Version20151202101336.php]: should not be ` NUMERIC(24, 10) ` ?

File: features/Context/catalog/footwear/attributes.yml [hunk 3]
```diff
@@ -263,6 +263,14 @@ attributes:
         sortOrder:   10
         metricFamily: Length
         defaultMetricUnit: CENTIMETER
+    volume:
+        labels:
+            en_US:
```
Comment (skeleton) [inline features/Context/catalog/footwear/attributes.yml]: add label please

File: upgrades/schema/Version20151202101336.php (hunk 2 above)
Comment (ClementGautier) [inline upgrades/schema/Version20151202101336.php]: yep probably

File: features/export/export_attributes.feature [hunk 4]
```diff
@@ -59,7 +60,7 @@ Feature: Export attributes
     pim_catalog_simpleselect;manufacturer;Manufacturer;;info;0;1;;;;;0;0
     pim_catalog_multiselect;weather_conditions;"Weather conditions";;info;0;1;;;;;0;0
     pim_catalog_textarea;description;Description;;info;0;1;;;;;1;1
-    pim_catalog_text;comment;Comment;other;;0;1;;;;;0;0
+    pim_catalog_text;comment;Comment;;other;0;1;;;;;0;0
```
Comment (phaseinducer) [inline features/export/export_attributes.feature]: Why did the values order change for this line ?

**Comment by jjanvier** [comment]
maybe I missread, but I didn't see a Behat that checks that the problem has been fixed

File: upgrades/schema/Version20151202101336.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+        $this->addSql('ALTER TABLE pim_catalog_metric CHANGE data data NUMERIC(24, 10) DEFAULT NULL, CHANGE base_data base_data NUMERIC(24, 10) DEFAULT NULL');
+    }
+
+    /**
+     * @param Schema $schema
+     */
+    public function down(Schema $schema)
+    {
+        throw new \RuntimeException('No revert is provided for the migrations.');
```
Comment (phaseinducer) [inline upgrades/schema/Version20151202101336.php]: `$this->throwIrreversibleMigrationException();`

File: upgrades/schema/Version20151202101336.php [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+/**
+ * Add precision to the Metric entity to be able to save data and baseData for numbers under 1E-4.
+ * We need this precision, for example, when we convert data in millimeter cube to meter cube (1E-8).
+ * For data which are already saved, as many 0 as needed will be added at the end of the number.
+ *
+ * @author    Willy Mesnage <willy.mesnage@akeneo.com>
+ * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class Version20151202101336 extends AbstractMigration
```
Comment (phaseinducer) [inline upgrades/schema/Version20151202101336.php]: Migration classes must be written in standard edition.

File: features/export/export_attributes.feature (hunk 4 above)
Comment (willy-ahva) [inline features/export/export_attributes.feature]: other is the group and not the label. Don't know why it was not broken before.

File: features/filter/metric.feature [hunk 7]
```diff
@@ -6,18 +6,24 @@ Feature: Filter on metric attributes
...
-      | CASSAVA   | 44 CENTIMETER |
-      | PEN       | 0 CENTIMETER  |
-      | FINGER    |               |
+      | sku       | length        | volume              |
+      | SUSHIROLL | 12 CENTIMETER |                     |
+      | OBELISK   | 35 CENTIMETER |                     |
+      | CASSAVA   | 44 CENTIMETER |                     |
+      | PEN       | 0 CENTIMETER  |                     |
+      | FINGER    |               |                     |
+      | MOUTH     |               | 12 CUBIC_MILLIMETER |
```
Comment (willy-ahva) [inline features/filter/metric.feature]: @jjanvier here is the behat which checks the bug

File: upgrades/schema/Version20151202101336.php (hunk 6 above)
Comment (jjanvier) [inline upgrades/schema/Version20151202101336.php]: not for master (we'll copy paste them at the end)

File: features/export/export_attributes.feature (hunk 4 above)
Comment (phaseinducer) [inline features/export/export_attributes.feature]: weird, ok

**Comment by phaseinducer** [comment]
GTM after fix

**Comment by skeleton** [comment]
gtm

File: upgrades/schema/Version20151202101336.php (hunk 5 above)
Comment (jjanvier) [inline upgrades/schema/Version20151202101336.php]: +1

**Comment by jjanvier** [comment]
gtm after fix

**Pull request merged**